woocommerce/woocommerce-ios
# [Order form] Add dedicated view model for `ProductDiscountView`

Closes: #11425
Part of #11357
⚠️ Depends on #11424 ⚠️ 


## Why
Up to now, `ProductDiscountView` was being driven by a mix of view models, including `ProductRowViewModel` and `ProductInOrderViewModel`. This adds a dedicated view model `ProductDiscountViewModel` to use with that view instead. This is another step toward disentangling the logic used in `CollapsibleProductCard` from `ProductRowViewModel`, eventually allowing the products section in the order form to be displayed using `OrderItem` data.

## How

* Adds a new `ProductDiscountViewModel` with the properties needed for `ProductDiscountView`. (Many of these properties are moved from `ProductInOrderViewModel`.)
* Updates `ProductDiscountView` to use `ProductDiscountViewModel`.
   * Note: `ProductDiscountViewModel` includes the property `discountDetailsViewModel`, but rather than using that nested view model within the view it's added as an `@ObservedObject` directly in `ProductDiscountView`. This seemed like the most straightforward way to initialize that view model and observe its changes, but let me know if you think there's a better way of doing that.
* Updates `EditableOrderViewModel` to create the required `ProductDiscountViewModel`. I also updated the naming in `EditableOrderViewModel` to hopefully make it clearer what is being done there.
* Removes `ProductInOrder` and `ProductInOrderViewModel`, since that view and view model are no longer used anywhere.

## Testing instructions

There should be no changes to the order form UX:

- Go to the Orders tab
- Tap on `+` to create an order
- Add at least one product to the order
- Expand any product card
- Tap "Add discount" --> Confirm the expected product details appear in the discount sheet
- Enter a fixed price or percentage discount --> Confirm the view updates to show the entered discount and price after discount
- Tap "Add" to save the discount --> Confirm the discount appears for the product in the order
- Tap "Create" to create the order --> Confirm the discount is reflected in the order


---
